src/Process/FeatureProcessing.js

```diff
@@ -0,0 +1,104 @@
+import ObjectRemovalHelper from './ObjectRemovalHelper';
+
+function applyOffset(obj, offset) {
+    if (obj.geometry) {
+        if (obj.geometry instanceof THREE.BufferGeometry) {
+            for (let i = 0; i < obj.geometry.attributes.position.count; i++) {
+                obj.geometry.attributes.position.array[3 * i] += offset.x;
+                obj.geometry.attributes.position.array[3 * i + 1] += offset.y;
+                obj.geometry.attributes.position.array[3 * i + 2] += offset.z;
+            }
+            obj.geometry.attributes.position.needsUpdate = true;
+        } else {
+            for (const v of obj.geometry.vertices) {
+                v.add(offset);
+            }
+            obj.geometry.verticesNeedUpdate = true;
+        }
+    }
+    obj.children.forEach(c => applyOffset(c, offset));
+}
+
+export default {
+    update(colorFunction) {
+        return function _(context, layer, node) {
+            if (!node.parent && node.children.length) {
+                // if node has been removed dispose three.js resource
+                ObjectRemovalHelper.removeChildrenAndCleanupRecursively(layer.id, node);
+                return;
+            }
+            if (!node.visible) {
+                return;
+            }
+
+            const features = node.children.filter(n => n.layer == layer.id);
+            if (features.length > 0) {
+                return features;
+            }
+
+            if (!layer.tileInsideLimit(node, layer)) {
+                return;
+            }
+
+            if (node.layerUpdateState[layer.id] === undefined) {
+                node.layerUpdateState[layer.id] = new LayerUpdateState();
+            }
+
+            const ts = Date.now();
+
+            if (!node.layerUpdateState[layer.id].canTryUpdate(ts)) {
+                return;
+            }
+
+            node.layerUpdateState[layer.id].newTry();
+
+            const command = {
+                layer,
+                view: context.view,
+                threejsLayer: layer.threejsLayer,
+                requester: node,
+            };
+
+            context.scheduler.execute(command).then((result) => {
+                // if request return empty json, WFS_Provider.getFeatures return undefined
+                if (result) {
+                    node.layerUpdateState[layer.id].success();
+                    if (!node.parent) {
+                        ObjectRemovalHelper.removeChildrenAndCleanupRecursively(layer.id, result);
+                        return;
+                    }
+                    // We don't use node.matrixWorld here, because feature coordinates are
+                    // expressed in crs coordinates (which may be different than world coordinates,
+                    // if node's layer is attached to an Object with a non-identity transformation)
+                    const tmp = node.extent.center().xyz().negate();
+                    applyOffset(result, tmp);
```

Comment:
Maybe we could return geometries (centred on (0, 0, 0)) with associated coordinates and create mesh here with the right transformation. And remove `applyOffset`